index.go

```diff
@@ -471,6 +491,78 @@ func (a activity) Render() []*html.Node {
 	return []*html.Node{htmlg.DivClass("activity", nodes...)}
 }
 
+func issueName(p *github.IssueCommentEvent) htmlg.Component {
+	n := iconLink{
+		Text:    shortTitle(*p.Issue.Title),
+		Tooltip: *p.Issue.Title,
+		URL:     *p.Comment.HTMLURL,
+		Black:   true,
+	}
+	switch *p.Issue.State {
+	case "open":
+		n.Icon = octiconssvg.IssueOpened
+		n.Color = RGB{R: 0x6c, G: 0xc6, B: 0x44} // Green.
+	case "closed":
+		n.Icon = octiconssvg.IssueClosed
+		n.Color = RGB{R: 0xbd, G: 0x2c, B: 0x00} // Red.
+	default:
+		log.Println("activity.Render: unsupported *github.IssueCommentEvent Issue.State:", *p.Issue.State)
+		n.Icon = octiconssvg.IssueOpened
+	}
+	return n
+}
+func prName(p *github.IssueCommentEvent) htmlg.Component {
+	n := iconLink{
+		Text:    shortTitle(*p.Issue.Title),
+		Tooltip: *p.Issue.Title,
+		URL:     *p.Comment.HTMLURL,
+		Black:   true,
+	}
+	switch *p.Issue.State {
+	case "open":
+		n.Icon = octiconssvg.GitPullRequest
+		n.Color = RGB{R: 0x6c, G: 0xc6, B: 0x44} // Green.
+	case "closed":
+		n.Icon = octiconssvg.GitPullRequest
+		n.Color = RGB{R: 0xbd, G: 0x2c, B: 0x00} // Red.
```

Comment:
Thanks for the suggestion. I have considered it, and I know with high certainty that this is a refactor that's not worth doing at this time. The benefits are very little.

The reason is that I have these red/green/purple colors indicating merge states (that map to GitHub colors) in quite a few places in my GOPATH. However, some of them use [`issues.RGB`](https://godoc.org/github.com/shurcooL/issues#RGB) type, some use [`notifications.RGB`](https://godoc.org/github.com/shurcooL/notifications#RGB). I have the `RGB` color type defined locally in a few places, including in this repo.

```
Searching 1427 files for "RGB{R: 0xbd, G: 0x2c, B: 0x00}"

/Users/Dmitri/Dropbox/Work/2013/GoLand/src/github.com/shurcooL/home/index.go:
  278  			case "closed":
  279  				details.Icon = octiconssvg.IssueClosed
  280: 				details.Color = RGB{R: 0xbd, G: 0x2c, B: 0x00} // Red.
  281  			case "reopened":
  282  				details.Icon = octiconssvg.IssueReopened
  ...
  306  				e.Action = component.Text("closed a pull request in")
  307  				details.Icon = octiconssvg.GitPullRequest
  308: 				details.Color = RGB{R: 0xbd, G: 0x2c, B: 0x00} // Red.
  309  			case *p.PullRequest.Merged:
  310  				e.Action = component.Text("merged a pull request in")
  ...
  505  	case "closed":
  506  		n.Icon = octiconssvg.IssueClosed
  507: 		n.Color = RGB{R: 0xbd, G: 0x2c, B: 0x00} // Red.
  508  	default:
  509  		log.Println("activity.Render: unsupported *github.IssueCommentEvent Issue.State:", *p.Issue.State)
  ...
  525  	case "closed":
  526  		n.Icon = octiconssvg.GitPullRequest
  527: 		n.Color = RGB{R: 0xbd, G: 0x2c, B: 0x00} // Red.
  528  	// TODO: Detect merged somehow? It's likely going to require making an API call.
  529  	default:
  ...
  546  	case p.PullRequest.MergedAt == nil && *p.PullRequest.State == "closed":
  547  		n.Icon = octiconssvg.GitPullRequest
  548: 		n.Color = RGB{R: 0xbd, G: 0x2c, B: 0x00} // Red.
  549  	case p.PullRequest.MergedAt != nil:
  550  		n.Icon = octiconssvg.GitMerge

/Users/Dmitri/Dropbox/Work/2013/GoLand/src/github.com/shurcooL/issues/fs/notifications.go:
  120  		return notifications.RGB{R: 0x6c, G: 0xc6, B: 0x44}
  121  	case issues.ClosedState: // Closed.
  122: 		return notifications.RGB{R: 0xbd, G: 0x2c, B: 0x00}
  123  	default:
  124  		return notifications.RGB{}

/Users/Dmitri/Dropbox/Work/2013/GoLand/src/github.com/shurcooL/notifications/githubapi/githubapi.go:
   83  			case err == nil && state == "closed":
   84  				notification.Icon = "issue-closed"
   85: 				notification.Color = notifications.RGB{R: 0xbd, G: 0x2c, B: 0x00} // Red.
   86  			default:
   87  				notification.Icon = "issue-opened"
   ..
  102  				notification.Color = notifications.RGB{R: 0x6c, G: 0xc6, B: 0x44} // Green.
  103  			case err == nil && state == "closed":
  104: 				notification.Color = notifications.RGB{R: 0xbd, G: 0x2c, B: 0x00} // Red.
  105  			case err == nil && state == "merged":
  106  				notification.Color = notifications.RGB{R: 0x6e, G: 0x54, B: 0x94} // Purple.

8 matches across 3 files
```

It's very easy to find all instances of these colors and keep them in sync. They change very infrequently (almost never).

Until I find a good place for that `RGB` type, I couldn't have a globally usable set of colors, I'd need to cast things anyway... And coming up with a new repo for `RGB` is a lot of work, etc.

So in the end, this is something I've thought about and will deal with later, it's not causing any downsides to leave it as is for now. Thanks for the idea though!